File: 0448-find-all-numbers-disappeared-in-an-array/solution.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        result = [0] * (len(nums) + 1)
        solution = []
        for i in range(len(nums)):
            result[nums[i]] = 1
        for j in range(1,len(nums) + 1):
            if not result[j]:
                solution.append(j)
        return solution

File: 0448-find-all-numbers-disappeared-in-an-array/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findDisappearedNumbers(int* nums, int numsSize, int* returnSize) {
    int i;
    for (i = 0; i < numsSize; i++) {
        nums[abs(nums[i]) - 1] = -abs(nums[abs(nums[i]) - 1]);
    }
    *returnSize = 0;
    for (i = 0; i < numsSize; i++) {
        if (nums[i] > 0)
            nums[(*returnSize)++] = i + 1;
    }
    return nums;
}
